pkg/controller/core/localqueue_controller.go

```diff
@@ -74,17 +77,49 @@ func (r *LocalQueueReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	ctx = ctrl.LoggerInto(ctx, log)
 	log.V(2).Info("Reconciling LocalQueue")
 
-	// Shallow copy enough for now.
-	oldStatus := queueObj.Status
+	oldStatus := queueObj.Status.DeepCopy()
 
 	pending, err := r.queues.PendingWorkloads(&queueObj)
 	if err != nil {
 		r.log.Error(err, "Failed to retrieve localQueue status")
 		return ctrl.Result{}, err
 	}
-
 	queueObj.Status.PendingWorkloads = pending
 	queueObj.Status.AdmittedWorkloads = r.cache.AdmittedWorkloadsInLocalQueue(&queueObj)
+
+	err = r.client.Get(ctx, client.ObjectKey{Name: string(queueObj.Spec.ClusterQueue)}, &kueue.ClusterQueue{})
+	if client.IgnoreNotFound(err) != nil {
+		return ctrl.Result{}, nil
+	}
+	if r.cache.ClusterQueueActive(string(queueObj.Spec.ClusterQueue)) {
+		meta.SetStatusCondition(&queueObj.Status.Conditions, metav1.Condition{
+			Type:    kueue.LocalQueueActive,
+			Status:  metav1.ConditionTrue,
+			Reason:  "Ready",
+			Message: "Can submit new workloads to clusterQueue",
+		})
+	} else {
+		var reason string
+		if errors.IsNotFound(err) {
+			reason = "ClusterQueueDoesNotExist"
+		} else {
+			reason = "ClusterQueueIsInactive"
+		}
+		//if r.queues.ClusterQueueExists(string(queueObj.Spec.ClusterQueue)) {
+		//	reason = "ClusterQueueIsInactive"
+		//} else {
+		//	reason = "ClusterQueueDoesNotExist"
+		//}
```

Comment:
@alculquicondor It seems we can not get the clusterQueue when we watch all events for clusterQueue instead of a notifier.

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/623/pull-kueue-test-integration-main/1635751835224510464

So we might need a notifier. WDYT?

Reply:
The controller could not get a clusterQueue referenced by the queue from either the client or the qManager.

Reply:
Sorry, this is my bad.

Reply:
I just remembered that I need to implement each event handler.